src/tools/ScenarioMeasurement/Startup/ParserUtility.cs

```diff
@@ -0,0 +1,94 @@
+﻿using Microsoft.Diagnostics.Tracing;
+using System;
+using System.Collections.Generic;
+using System.CommandLine;
+using System.Data;
+using System.Data.Common;
+using System.Diagnostics;
+using System.Text.RegularExpressions;
+
+namespace ScenarioMeasurement
+{
+    class ParserUtility
+    {
+
+        public static bool MatchProcessStart(TraceEvent evt, TraceSourceManager source, string processName, IList<int> pids, string commandLine)
+        {
+            return MatchCommandLine(evt, source, commandLine) &&
+                   MatchProcessName(evt, source, processName) &&
+                   MatchProcessID(evt, source, pids);
+        }
+
+        public static bool MatchCommandLine(TraceEvent evt, TraceSourceManager source, string commandLine)
+        {
+            if (source.IsWindows)
+            {
+                if (((string)GetPayloadValue(evt, "CommandLine")).Trim() != commandLine)
+                {
+                    return false;
+                }
+            }
+            // Match the command line as well because pids might be reused during the session
+            return true;
+        }
+
+        public static bool MatchProcessName(TraceEvent evt, TraceSourceManager source, string processName)
+        {
+            if (source.IsWindows)
+            {
+                if (!processName.Equals(evt.ProcessName, StringComparison.OrdinalIgnoreCase))
+                {
+                    return false;
+                }
+            }
+            else
+            {
+                // 15 characters is the maximum length of a process name in Linux kernel event payload
+                if (processName.Length < 15)
+                {
+                    if (!processName.Equals(evt.ProcessName, StringComparison.OrdinalIgnoreCase))
+                    {
+                        return false;
+                    }
+                }
+                else if (!processName.Substring(0, 15).Equals(evt.ProcessName, StringComparison.OrdinalIgnoreCase))
+                {
+                    return false;
+                }
+            }
+            return true;
+        }
```

Comment:
If using an if/else block I tend to like having each block do it's own returning so that I can look at each block and know what will happen before moving on to the next one. In this case I have to look all the way at the end of the function to know what the code will do.

Reply:
hmm not sure how the first approach works? My logic is return false upon mismatch, and return true in the end when everything passed the match.

Reply:
It would look like the change below, but I think this is probably more of a personal preference. Keep it in a way that makes sense to you.
```suggestion
        public static bool MatchProcessName(TraceEvent evt, TraceSourceManager source, string processName)
        {
            if (source.IsWindows)
            {
                if (!processName.Equals(evt.ProcessName, StringComparison.OrdinalIgnoreCase))
                {
                    return false;
                }
                else
                {
                    return true;
                }
            }
            else
            {
                // 15 characters is the maximum length of a process name in Linux kernel event payload
                if (processName.Length < 15)
                {
                    if (!processName.Equals(evt.ProcessName, StringComparison.OrdinalIgnoreCase))
                    {
                        return false;
                    }
                }
                else if (!processName.Substring(0, 15).Equals(evt.ProcessName, StringComparison.OrdinalIgnoreCase))
                {
                    return false;
                }
                else
                {
                    return true;
                }
            }
        }
```